app/models/person.rb

```diff
@@ -237,11 +237,15 @@ def name(opts = {})
     if self.profile.nil?
       fix_profile
     end
-    @name ||= Person.name_from_attrs(self.profile.first_name, self.profile.last_name, self.diaspora_handle)
+    @name ||= Person.name_from_attrs(self.profile.full_name, self.diaspora_handle)
   end
 
-  def self.name_from_attrs(first_name, last_name, diaspora_handle)
-    first_name.blank? && last_name.blank? ? diaspora_handle : "#{first_name.to_s.strip} #{last_name.to_s.strip}".strip
+  def full_name
```

Comment:
You created this method in the `Person` model, while there already is a `full_name` attribute in the `Profile` model. I think this is really confusing, because you confused yourself and used `profile.full_name` where you probably wanted to use `person.full_name`.